controllers/auth.js

```diff
@@ -15,14 +16,27 @@ async function postRegister(req, res, next) {
 
 async function postLogin(req, res, next) {
     try {
-        const token = await login(req)
+        const user = await login(req)
         res.status(StatusCodes.OK).json({
-            token
+            user
         })
     } catch (error) {
         res.status(StatusCodes.UNAUTHORIZED).json({
             error: error.message
         })
     }
 }
-module.exports = { postRegister, postLogin }
+
+async function validateToken(req, res, next) {
+    try {
+        authenticateUser(req, res)
+        const user = await User.findById(req.user.userId)
```

Comment:
also this will send back the password, please remove the password,. Or create a new user from the fields and send back user